packages/front-end/components/Settings/EditDataSource/DataSourceInlineEditIdentityJoins/DataSourceInlineEditIdentityJoins.tsx

```diff
@@ -0,0 +1,203 @@
+import React, { FC, useCallback, useState } from "react";
+import { DataSourceQueryEditingModalBaseProps } from "../types";
+import { FaChevronRight, FaPencilAlt, FaPlus } from "react-icons/fa";
+import Code from "../../../Code";
+import MoreMenu from "../../../Dropdown/MoreMenu";
+import DeleteButton from "../../../DeleteButton";
+import cloneDeep from "lodash/cloneDeep";
+import {
+  DataSourceInterfaceWithParams,
+  IdentityJoinQuery,
+} from "back-end/types/datasource";
+import { AddEditIdentityJoinModal } from "./AddEditIdentityJoinModal";
+import Tooltip from "../../../Tooltip";
+
+type DataSourceInlineEditIdentityJoinsProps = DataSourceQueryEditingModalBaseProps;
+
+export const DataSourceInlineEditIdentityJoins: FC<DataSourceInlineEditIdentityJoinsProps> = ({
+  dataSource,
+  onSave,
+  onCancel,
+}) => {
+  const [uiMode, setUiMode] = useState<"view" | "edit" | "add">("view");
+  const [editingIndex, setEditingIndex] = useState<number>(-1);
+
+  const [openIndexes, setOpenIndexes] = useState<boolean[]>([]);
+
+  const handleCancel = useCallback(() => {
+    setUiMode("view");
+    setEditingIndex(-1);
+    onCancel();
+  }, [onCancel]);
+
+  const handleExpandCollapseForIndex = useCallback(
+    (index) => () => {
+      const currentValue = openIndexes[index] || false;
+      const updatedOpenIndexes = [...openIndexes];
+      updatedOpenIndexes[index] = !currentValue;
+
+      setOpenIndexes(updatedOpenIndexes);
+    },
+    [openIndexes]
+  );
+
+  const userIdTypes = dataSource.settings?.userIdTypes || [];
+  const addIsDisabled = userIdTypes.length < 2;
+  const identityJoins = dataSource?.settings?.queries?.identityJoins || [];
+
+  const handleAdd = useCallback(() => {
+    setUiMode("add");
+    setEditingIndex(identityJoins.length);
+  }, [identityJoins]);
+
+  const handleActionEditClicked = useCallback(
+    (idx: number) => () => {
+      setEditingIndex(idx);
+      setUiMode("edit");
+    },
+    []
+  );
+
+  const handleActionDeleteClicked = useCallback(
+    (idx: number) => async () => {
+      const copy = cloneDeep<DataSourceInterfaceWithParams>(dataSource);
+
+      copy.settings.queries.identityJoins.splice(idx, 1);
+
+      onSave(copy);
+    },
+    [identityJoins, onSave, dataSource]
+  );
+
+  const handleSave = useCallback(
+    (idx: number) => (identityJoin: IdentityJoinQuery) => {
+      const copy = cloneDeep<DataSourceInterfaceWithParams>(dataSource);
+      copy.settings.queries.identityJoins[idx] = identityJoin;
+      onSave(copy);
+    },
+    [dataSource, onSave, uiMode]
+  );
+
+  if (!dataSource) {
+    console.error("ImplementationError: dataSource cannot be null");
+    return null;
+  }
+
+  return (
+    <div className="">
+      {/* region Heading */}
+      {identityJoins.length > 0 || userIdTypes.length >= 2 ? (
+        <div className="d-flex justify-content-between align-items-center mb-3">
+          <div className="d-flex mt-2">
+            <h3>Identifier Join Tables</h3>
```

Comment:
I would change this title to just `Join Tables`.  When shown on the data source page, seeing "Identifier" 3 headings in a row makes it harder to understand what each section actually means.